### DISCUSSION TITLE
[Question]: Using Google Search and other online search plugins, the response speed is very slow.

### DISCUSSION THREAD

**Discussion by joeaki1983** [danny-avila/LibreChat#2041] [Troubleshooting]
### What is your question?

‌‌‌‌‌When using search plugins like Google Search, the response time is quite slow, usually taking about 1 minute to produce results. I am using the Azure OpenAI API and have ruled out any issues with the API. When not using web search plugins, the response speed is very fast. May I ask what could be causing this and how to solve it? Thank you.

### More Details

‌‌‌‌‌‌No more information.

### What is the main subject of your question?

User System/OAuth

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
This isn't really an issue but rather a product of the current implementation. The "Plugins" endpoint handles functions without streaming enabled, which produces this "slower" response.

In short, in order to parse both input and output for functions, there were many constraints with streaming and in many cases was not feasible with the framework it was first built with (langchain).

Now that a lot of these problems have been solved in langchain and other libraries, I will work on updating the whole system with a new approach that also allows streaming for "responsiveness."

**Reply by joeaki1983** [reply]
‌Understood, thank you for your reply. Looking forward to the update.
